From a8f701827e7f05dee59cb09a68d7ae374fc18ce0 Mon Sep 17 00:00:00 2001
From: Minmin Gong <gongminmin@msn.com>
Date: Sun, 20 Jan 2019 20:21:33 -0800
Subject: [PATCH 2/3] Fix for MinGW

---
 Modules/posixmodule.c | 14 +++++++-------
 Python/dynload_win.c  |  2 +-
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/Modules/posixmodule.c b/Modules/posixmodule.c
index 9ac021f6c5..c6410fceeb 100644
--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -337,8 +337,8 @@ corresponding Unix manual entries for more information on calls.");
 #  define HAVE_SYSTEM     1
 #  include <process.h>
 #else
-#  ifdef _MSC_VER
-     /* Microsoft compiler */
+#  if defined(_MSC_VER) || defined(__MINGW32__)
+     /* Microsoft compiler or MinGW */
 #    define HAVE_GETPPID    1
 #    define HAVE_GETLOGIN   1
 #    define HAVE_SPAWNV     1
@@ -350,7 +350,7 @@ corresponding Unix manual entries for more information on calls.");
 #    define HAVE_CWAIT      1
 #    define HAVE_FSYNC      1
 #    define fsync _commit
-#  endif  /* _MSC_VER */
+#  endif  /* _MSC_VER || __MINGW32__ */
 #endif  /* ! __WATCOMC__ || __QNX__ */
 
 /*[clinic input]
@@ -359,7 +359,7 @@ module os
 [clinic start generated code]*/
 /*[clinic end generated code: output=da39a3ee5e6b4b0d input=94a0f0f978acae17]*/
 
-#ifndef _MSC_VER
+#if !defined(_MSC_VER) && !defined(__MINGW32__)
 
 #if defined(__sgi)&&_COMPILER_VERSION>=700
 /* declare ctermid_r if compiling with MIPSPro 7.x in ANSI C mode
@@ -367,7 +367,7 @@ module os
 extern char        *ctermid_r(char *);
 #endif
 
-#endif /* !_MSC_VER */
+#endif /* !_MSC_VER && !__MINGW32__ */
 
 #if defined(__VXWORKS__)
 #  include <vxCpuLib.h>
@@ -428,7 +428,7 @@ extern char        *ctermid_r(char *);
 #  endif
 #endif
 
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(__MINGW32__)
 #  ifdef HAVE_DIRECT_H
 #    include <direct.h>
 #  endif
@@ -439,7 +439,7 @@ extern char        *ctermid_r(char *);
 #    include <process.h>
 #  endif
 #  include <malloc.h>
-#endif /* _MSC_VER */
+#endif /* _MSC_VER || __MINGW32__ */
 
 #ifndef MAXPATHLEN
 #  if defined(PATH_MAX) && PATH_MAX > 1024
diff --git a/Python/dynload_win.c b/Python/dynload_win.c
index 5dc4095237..1a8582734e 100644
--- a/Python/dynload_win.c
+++ b/Python/dynload_win.c
@@ -170,7 +170,7 @@ static char *GetPythonImport (HINSTANCE hModule)
    Return whether the DLL was found.
 */
 extern HMODULE PyWin_DLLhModule;
-static int
+int
 _Py_CheckPython3(void)
 {
     static int python3_checked = 0;
-- 
2.43.0.windows.1

