From 26ee6507d41a345f0706ec70016c358389bb81c9 Mon Sep 17 00:00:00 2001
From: Minmin Gong <gongminmin@msn.com>
Date: Mon, 15 Jan 2024 23:53:57 -0800
Subject: [PATCH 08/10] Update pyconfig.h for mingw

---
 cmake/config-mingw/pyconfig.h | 62 +++++++++++++++++++++++++++++------
 1 file changed, 52 insertions(+), 10 deletions(-)

diff --git a/cmake/config-mingw/pyconfig.h b/cmake/config-mingw/pyconfig.h
index 5af0378..a208503 100644
--- a/cmake/config-mingw/pyconfig.h
+++ b/cmake/config-mingw/pyconfig.h
@@ -84,18 +84,24 @@ WIN32 is still required for the locale module.
 #endif
 
 /* set the version macros for the windows headers */
-#ifdef MS_WINX64
-/* 64 bit only runs on XP or greater */
-#  define Py_WINVER _WIN32_WINNT_WINXP
-#  define Py_NTDDI NTDDI_WINXP
+#if PY_VERSION_HEX >= 0x03050000
+/* Python 3.5+ requires Windows Vista or greater */
+#  define Py_WINVER 0x0600 /* _WIN32_WINNT_VISTA */
+#  define Py_NTDDI NTDDI_VISTA
 #else
-/* Python 2.6+ requires Windows 2000 or greater */
-#  ifdef _WIN32_WINNT_WIN2K
-#    define Py_WINVER _WIN32_WINNT_WIN2K
+#  ifdef MS_WINX64
+/* 64 bit only runs on XP or greater */
+#    define Py_WINVER _WIN32_WINNT_WINXP
+#    define Py_NTDDI NTDDI_WINXP
 #  else
-#    define Py_WINVER 0x0500
+/* Python 2.6+ requires Windows 2000 or greater */
+#    ifdef _WIN32_WINNT_WIN2K
+#      define Py_WINVER _WIN32_WINNT_WIN2K
+#    else
+#      define Py_WINVER 0x0500
+#    endif
+#    define Py_NTDDI NTDDI_WIN2KSP4
 #  endif
-#  define Py_NTDDI NTDDI_WIN2KSP4
 #endif
 
 /* We only set these values when building Python - we don't want to force
@@ -124,7 +130,9 @@ WIN32 is still required for the locale module.
 #define Py_IS_NAN _isnan
 #define Py_IS_INFINITY(X) (!_finite(X) && !_isnan(X))
 #define Py_IS_FINITE(X) _finite(X)
-#define copysign _copysign
+#if PY_VERSION_HEX < 0x03060000
+#  define copysign _copysign
+#endif
 
 #include <basetsd.h>
 
@@ -222,6 +230,8 @@ Py_NO_ENABLE_SHARED to find out.  Also support MS_NO_COREDLL for b/w compat */
 
 #endif /* MS_WIN32 */
 
+#if PY_VERSION_HEX < 0x03060000
+
 /* define signed and unsigned exact-width 32-bit and 64-bit types, used in the
    implementation of Python long integers. */
 #ifndef PY_UINT32_T
@@ -258,6 +268,8 @@ Py_NO_ENABLE_SHARED to find out.  Also support MS_NO_COREDLL for b/w compat */
 #endif
 #endif
 
+#endif
+
 /* Fairly standard from here! */
 
 /* Define to 1 if you have the `copysign' function. */
@@ -377,6 +389,8 @@ Py_NO_ENABLE_SHARED to find out.  Also support MS_NO_COREDLL for b/w compat */
 /* Define if you want to use the GNU readline library */
 /* #define WITH_READLINE 1 */
 
+#if PY_VERSION_HEX < 0x03060000
+
 /* Define if you want to have a Unicode type. */
 #define Py_USING_UNICODE
 
@@ -384,6 +398,8 @@ Py_NO_ENABLE_SHARED to find out.  Also support MS_NO_COREDLL for b/w compat */
 /* This is enough for unicodeobject.h to do the "right thing" on Windows. */
 #define Py_UNICODE_SIZE 2
 
+#endif
+
 /* Use Python's own small-block memory-allocator. */
 #define WITH_PYMALLOC 1
 
@@ -528,6 +544,11 @@ Py_NO_ENABLE_SHARED to find out.  Also support MS_NO_COREDLL for b/w compat */
 /* Define if the compiler provides a wchar.h header file. */
 #define HAVE_WCHAR_H 1
 
+#if PY_VERSION_HEX >= 0x03060000
+/* The size of `wchar_t', as computed by sizeof. */
+#define SIZEOF_WCHAR_T 2
+#endif
+
 /* Define if you have the dl library (-ldl).  */
 /* #undef HAVE_LIBDL */
 
@@ -563,4 +584,25 @@ Py_NO_ENABLE_SHARED to find out.  Also support MS_NO_COREDLL for b/w compat */
    least significant byte first */
 #define DOUBLE_IS_LITTLE_ENDIAN_IEEE754 1
 
+#if PY_VERSION_HEX >= 0x03070000
+/* Define to 1 if you have the `erf' function. */
+#define HAVE_ERF 1
+
+/* Define to 1 if you have the `erfc' function. */
+#define HAVE_ERFC 1
+
+/* Define if you have the 'inet_pton' function. */
+#define HAVE_INET_PTON 1
+
+/* framework name */
+#define _PYTHONFRAMEWORK ""
+
+/* Define if libssl has X509_VERIFY_PARAM_set1_host and related function */
+#define HAVE_X509_VERIFY_PARAM_SET1_HOST 1
+#endif
+
+#if PY_VERSION_HEX >= 0x03090000
+#define PLATLIBDIR "lib"
+#endif
+
 #endif /* !Py_CONFIG_H */
-- 
2.43.0.windows.1

