From 9ae8acbcdaad00499fcbf6f8dfb7e85447afd852 Mon Sep 17 00:00:00 2001
From: Minmin Gong <gongminmin@msn.com>
Date: Thu, 2 Apr 2020 21:07:19 -0700
Subject: [PATCH] Fix operator== on wstring_view and wchar_t raw string

---
 include/nonstd/string_view.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/nonstd/string_view.hpp b/include/nonstd/string_view.hpp
index 808a276..7ee2627 100644
--- a/include/nonstd/string_view.hpp
+++ b/include/nonstd/string_view.hpp
@@ -1010,12 +1010,12 @@ nssv_constexpr bool operator>= (
 template< class CharT, class Traits>
 nssv_constexpr bool operator==(
     basic_string_view<CharT, Traits> lhs,
-    char const * rhs ) nssv_noexcept
+    CharT const * rhs ) nssv_noexcept
 { return lhs.compare( rhs ) == 0; }
 
 template< class CharT, class Traits>
 nssv_constexpr bool operator==(
-    char const * lhs,
+    CharT const * lhs,
     basic_string_view<CharT, Traits> rhs ) nssv_noexcept
 { return rhs.compare( lhs ) == 0; }
 
-- 
2.17.1.windows.2

