From 4dc0e7a915240157dff6df8af8fc9ff58c070c3d Mon Sep 17 00:00:00 2001
From: Minmin Gong <gongminmin@msn.com>
Date: Tue, 16 Feb 2021 20:08:22 -0800
Subject: [PATCH 3/3] Export the animation name to gltf2

---
 code/AssetLib/glTF2/glTF2Exporter.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/code/AssetLib/glTF2/glTF2Exporter.cpp b/code/AssetLib/glTF2/glTF2Exporter.cpp
index 565117d..dbe7ec2 100644
--- a/code/AssetLib/glTF2/glTF2Exporter.cpp
+++ b/code/AssetLib/glTF2/glTF2Exporter.cpp
@@ -1386,6 +1386,7 @@ void glTF2Exporter::ExportAnimations()
             nameAnim = anim->mName.C_Str();
         }
         Ref<Animation> animRef = mAsset->animations.Create(nameAnim);
+        animRef->name = nameAnim;
 
         for (unsigned int channelIndex = 0; channelIndex < anim->mNumChannels; ++channelIndex) {
             const aiNodeAnim* nodeChannel = anim->mChannels[channelIndex];
-- 
2.17.1.windows.2

