From 0b26dacfbcab8bdb607ba112a6bdc77bbf766676 Mon Sep 17 00:00:00 2001
From: Minmin Gong <gongminmin@msn.com>
Date: Sun, 22 Nov 2020 20:33:21 -0800
Subject: [PATCH] Fix SAL issues in GCC 10

---
 Alc/backends/dsound.cpp | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/Alc/backends/dsound.cpp b/Alc/backends/dsound.cpp
index 93de3135..663d8645 100644
--- a/Alc/backends/dsound.cpp
+++ b/Alc/backends/dsound.cpp
@@ -47,6 +47,33 @@
 
 /* MinGW-w64 needs this for some unknown reason now. */
 using LPCWAVEFORMATEX = const WAVEFORMATEX*;
+#ifdef __MINGW32__
+#   include <sal.h>
+#   ifndef __in
+#      define __in
+#   endif
+#   ifndef __in_bcount_opt
+#      define __in_bcount_opt(size)
+#   endif
+#   ifndef __in_ecount_opt
+#      define __in_ecount_opt(size)
+#   endif
+#   ifndef __out
+#      define __out
+#   endif
+#   ifndef __out_bcount_opt
+#      define __out_bcount_opt(size)
+#   endif
+#   ifndef __out_ecount_opt
+#      define __out_ecount_opt(size)
+#   endif
+#   ifndef __null
+#      define __null
+#   endif
+#   ifndef __deref_out_bcount
+#      define __deref_out_bcount(size)
+#   endif
+#endif
 #include <dsound.h>
 
 
-- 
2.17.1.windows.2

