From 127a264d4118d80eeac4e858cd8fdad0bc1b361f Mon Sep 17 00:00:00 2001
From: Minmin Gong <gongminmin@msn.com>
Date: Sun, 20 Jan 2019 20:21:33 -0800
Subject: [PATCH 2/3] Fix for MinGW

---
 Modules/posixmodule.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Modules/posixmodule.c b/Modules/posixmodule.c
index 726e3723f9..e67782aa74 100644
--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -181,7 +181,7 @@ corresponding Unix manual entries for more information on calls.");
 #define HAVE_SYSTEM     1
 #include <process.h>
 #else
-#ifdef _MSC_VER         /* Microsoft compiler */
+#if defined(_MSC_VER) || defined(__MINGW32__)         /* Microsoft compiler or MinGW */
 #define HAVE_GETPPID    1
 #define HAVE_GETLOGIN   1
 #define HAVE_SPAWNV     1
@@ -213,7 +213,7 @@ corresponding Unix manual entries for more information on calls.");
 #define HAVE_SYSTEM     1
 #define HAVE_WAIT       1
 #define HAVE_TTYNAME    1
-#endif  /* _MSC_VER */
+#endif  /* _MSC_VER || __MINGW32__ */
 #endif  /* ! __WATCOMC__ || __QNX__ */
 
 
@@ -292,7 +292,7 @@ extern char        *ctermid_r(char *);
 #endif
 #endif
 
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(__MINGW32__)
 #ifdef HAVE_DIRECT_H
 #include <direct.h>
 #endif
@@ -314,7 +314,7 @@ extern char        *ctermid_r(char *);
 #include <shellapi.h>   /* for ShellExecute() */
 #include <lmcons.h>     /* for UNLEN */
 #define HAVE_SYMLINK
-#endif /* _MSC_VER */
+#endif /* _MSC_VER || __MINGW32__ */
 
 #ifndef MAXPATHLEN
 #if defined(PATH_MAX) && PATH_MAX > 1024
-- 
2.17.1.windows.2

