From 44a48402469b9440532cb688f73e8069ac84bf1f Mon Sep 17 00:00:00 2001
From: Minmin Gong <gongminmin@msn.com>
Date: Sat, 11 May 2019 16:10:22 -0700
Subject: [PATCH 5/7] Fix missing __mingw_uuidof_s problem on gcc 8.3

---
 Source/LibJXR/common/include/guiddef.h | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/Source/LibJXR/common/include/guiddef.h b/Source/LibJXR/common/include/guiddef.h
index d769c5cc..6b62a531 100644
--- a/Source/LibJXR/common/include/guiddef.h
+++ b/Source/LibJXR/common/include/guiddef.h
@@ -53,10 +53,15 @@ typedef struct _GUID {
 #endif
 #endif
 
-#if defined(__MINGW32__) && defined(__cplusplus) && (USE___UUIDOF == 0)
-extern "C++" {
-__extension__ template<typename T> const GUID &__mingw_uuidof();
-}
+#if defined(__MINGW32__) && defined(__cplusplus) && (USE___UUIDOF == 0)
+extern "C++" {
+#if __cpp_constexpr >= 200704l && __cpp_inline_variables >= 201606L
+__extension__ template<typename T> struct __mingw_uuidof_s;
+__extension__ template<typename T> constexpr const GUID &__mingw_uuidof();
+#else
+__extension__ template<typename T> const GUID &__mingw_uuidof();
+#endif
+}
 #endif
 
 #ifndef FAR
-- 
2.17.1.windows.2

